lib/network_utils.ml

```diff
@@ -208,6 +215,36 @@ module Sysfs = struct
 		|> (fun p -> try read_one_line p |> duplex_of_string with _ -> Duplex_unknown)
 		in (speed, duplex)
 
+	let get_dev_nums_with_same_driver driver = 
+		try
+			Sys.readdir ("/sys/bus/pci/drivers/" ^ driver)
+			|> Array.to_list
+			|> List.filter (Re.execp (Re_perl.compile_pat "\d+:\d+:\d+\.\d+"))
+			|> List.length
+		with _ -> 0
+
+	let parent_device_of_vf pcibuspath =
+		try
+			let pf_net_path = Printf.sprintf "/sys/bus/pci/devices/%s/physfn/net" pcibuspath in
+			let devices = Sys.readdir pf_net_path in
+			if Array.length devices = 1 then devices.(0)
+			else raise (Sys_error ("Can not get parent device for " ^ pcibuspath))
+		with _ -> raise (Sys_error ("Can not get parent device for " ^ pcibuspath))
```

Comment:
The `Sys_error` exception is raised by functions in the `Sys` module. I don't think that is appropriate here. Also, this exception will bubble up all the way to the top. I'd prefer to return an option type from functions like this, so that the control flow is easier to understand. Or even better, use the `result` type and monadic operators from `Rresult`.

Note that generally speaking, the error handling in networkd is quite poor. But it would be good to start right now to make it better in the new code.